### PR TITLE
feat: bump up the go version

### PR DESCRIPTION
**Pull request by Aju100**

### CODE REVIEW

**Comment by Aju100** [comment]
Fixed the jx-helpers one https://github.com/jenkins-x/jx/issues/8283

Seems like there need to be version upgrade of API and other dependencies too @ankitm123 Correct me if I am wrong here

File: Makefile [hunk 1]
```diff
@@ -13,7 +13,7 @@ ORG := jenkins-x
 ORG_REPO := $(ORG)/$(NAME)
 RELEASE_ORG_REPO := $(ORG_REPO)
 ROOT_PACKAGE := github.com/$(ORG_REPO)
-GO_VERSION := 1.17.9
+GO_VERSION := 1.18
```
Comment (ankitm123) [inline Makefile:16]: change to 1.18.4

**Review by ankitm123 — COMMENTED** [review]
Just one minor thing

File: Makefile (hunk 1 above)
Comment (Aju100) [reply Makefile]: done @ankitm123

**Comment by ankitm123** [comment]
@Aju100 now that kube-client has been upgraded, I think you can also bump the version in go.mod (and run go mod tidy). Saves us one extra PR :grimacing:

**Comment by ankitm123** [comment]
Also change commit message to `refactor` instead of `feat` - we are not adding any changes to the external facing API.

**Pull request merged**